pkg/input/remoteread/remoteread.go

```diff
@@ -1,3 +1,219 @@
 package remoteread
 
 // TBD, help wanted!
+import (
+	"context"
+	"github.com/go-kit/kit/log"
+	config_util "github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/prometheus/prometheus/storage/remote"
+	"github.com/prometheus/prometheus/tsdb/chunkenc"
+	"github.com/thanos-community/obslytics/pkg/input"
+	"net/url"
+
+	"time"
+)
+
+type remoteReadInput struct {
+	logger log.Logger
+	conf   input.InputConfig
+}
+
+func NewRemoteReadInput(logger log.Logger, conf input.InputConfig) (remoteReadInput, error) {
+	return remoteReadInput{logger: logger, conf: conf}, nil
+}
+
+func (i remoteReadInput) Open(ctx context.Context, params input.SeriesParams) (input.SeriesIterator, error) {
+
+	tlsConfig := config_util.TLSConfig{
+		CAFile:             i.conf.TLSConfig.CAFile,
+		CertFile:           i.conf.TLSConfig.CertFile,
+		KeyFile:            i.conf.TLSConfig.KeyFile,
+		ServerName:         i.conf.TLSConfig.ServerName,
+		InsecureSkipVerify: i.conf.TLSConfig.InsecureSkipVerify,
+	}
+
+	httpConfig := config_util.HTTPClientConfig{
+		//BasicAuth:       nil,
```

Comment:
Commented code? ):

Reply:
This was just to show what other http config params we could add in the config